tests/test_georaster.py

```diff
@@ -795,3 +795,9 @@ def test_resampling_str(self):
         img3a = img1.reproject(img2, resampling="q1")
         img3b = img1.reproject(img2, resampling=rio.warp.Resampling.q1)
         assert img3a == img3b
+        
+    def test_polygonize(self):
+        """Test that polygonize doesn't raise errors."""
+        img = gr.Raster(datasets.get_path('landsat_B4'))
+        assert gdf = img.polygonize()
```

Comment:
This assertion fails because it's not an assertion; it's an assignment. Also, an instantiated class is not equal to the class definition, so `isinstance` has to be used.
It should rather be:
```python
assert isinstance(img.polygonize(), gdf)
```